test/Microsoft.OpenApi.Tests/Services/OpenApiFilterServiceTests.cs

```diff
@@ -19,27 +19,37 @@ public OpenApiFilterServiceTests()
         }
 
         [Theory]
-        [InlineData("users.user.ListUser")]
-        [InlineData("users.user.GetUser")]
-        [InlineData("administrativeUnits.restore")]
-        [InlineData("graphService.GetGraphService")]
-        public void ReturnFilteredOpenApiDocumentBasedOnOperationIds(string operationId)
+        [InlineData("users.user.ListUser", null)]
+        [InlineData("users.user.GetUser", null)]
+        [InlineData("administrativeUnits.restore", null)]
+        [InlineData("graphService.GetGraphService", null)]
+        [InlineData(null, "users.user")]
+        [InlineData(null, "applications.application")]
+        [InlineData(null, "reports.Functions")]
+        public void ReturnFilteredOpenApiDocumentBasedOnOperationIds(string operationIds, string tags)
         {
             // Act
-            var predicate = OpenApiFilterService.CreatePredicate(operationId);
+            var predicate = OpenApiFilterService.CreatePredicate(operationIds, tags);
             var subsetOpenApiDocument = OpenApiFilterService.CreateFilteredDocument(_openApiDocumentMock, predicate);
 
             // Assert
             Assert.NotNull(subsetOpenApiDocument);
-            Assert.Single(subsetOpenApiDocument.Paths);
+            if (!string.IsNullOrEmpty(operationIds))
+            {
+                Assert.Single(subsetOpenApiDocument.Paths);
+            }
+            else if (!string.IsNullOrEmpty(tags))
+            {
+                Assert.NotEmpty(subsetOpenApiDocument.Paths);
+            }
         }
 
         [Fact]
         public void ThrowsInvalidOperationExceptionInCreatePredicateWhenInvalidOperationIdIsSpecified()
         {
             // Act and Assert
-            var message = Assert.Throws<InvalidOperationException>(() =>OpenApiFilterService.CreatePredicate(null)).Message;
-            Assert.Equal("OperationId needs to be specified.", message);
+            var message = Assert.Throws<InvalidOperationException>(() =>OpenApiFilterService.CreatePredicate(null, null)).Message;
```

Comment:
Should we also test for whether an `Exception` is thrown when both the `operationIds` and `tags` are specified?

Reply:
Yeah I can add that too.